src/components/chatcanvas/ChatBox.jsx

```diff
@@ -1,10 +1,274 @@
+import { useEffect, useRef, useState } from "react";
 import Textarea from "./TextareaChatCanvas";
+import axios from "axios";
+import { getUser } from "../../utils/authHandler";
+import { io } from "socket.io-client";
```

Comment:
Use the `socket` variable exported from `globals.js`, so it doesn't open two socket connections.

Reply:
code fixed. thanks for catching that!

Reply:
> Use the `socket` variable exported from `globals.js`, so it doesn't open two socket connections.

code fixed. Thanks for catching that!